cms/server/contest/handlers/__init__.py

```diff
@@ -97,6 +102,12 @@
     (r"/communication", CommunicationHandler),
     (r"/question", QuestionHandler),
 
+    # API
+    (r"/api/login", ApiLoginHandler),
+    (r"/api/task_list", ApiTaskListHandler),
+    (r"/api/(.*)/submit", ApiSubmitHandler),
+    (r"/api/(.*)/submission_list", ApiSubmissionListHandler),
```

Comment:
I suspect that `(.*)` with eat even a slash. It would be better to use a narrower regex.